app/main/util/search_utils.py

```diff
@@ -0,0 +1,174 @@
+import opensearchpy
+from flask import abort, current_app, request
+from opensearchpy import OpenSearch, RequestsHttpConnection
+
+from app.main.util.filter_sort_builder import build_sorting_orders_open_search
+from app.main.util.pagination import calculate_total_pages, get_pagination
+
+
+def get_param(param):
+    return request.form.get(param, "") or request.args.get(param, "")
+
+
+def get_query_and_search_area():
+    """Fetch query and search_area from form or request args"""
+    query = get_param("query")
+    search_area = get_param("search_area")
+    return query.strip(), search_area
+
+
+def setup_opensearch():
+    """Setup and return an OpenSearch client"""
+    return OpenSearch(
+        hosts=current_app.config.get("OPEN_SEARCH_HOST"),
+        http_auth=current_app.config.get("OPEN_SEARCH_HTTP_AUTH"),
+        use_ssl=True,
+        verify_certs=True,
+        ca_certs=current_app.config.get("OPEN_SEARCH_CA_CERTS"),
+        connection_class=RequestsHttpConnection,
+    )
+
+
+def execute_search(open_search, dsl_query, page, per_page):
+    """Execute the search query using OpenSearch"""
+    size = per_page
+    from_ = size * (page - 1)
+    try:
+        return open_search.search(
+            dsl_query,
+            from_=from_,
+            size=size,
+            timeout=current_app.config["OPEN_SEARCH_TIMEOUT"],
+        )
+    except opensearchpy.exceptions.ConnectionTimeout:
+        abort(504)
+
+
+def get_pagination_info(results, page, per_page):
+    """Calculate pagination information"""
+    total_records = (
+        results["hits"]["total"]["value"] if "hits" in results else 0
+    )
+    page_count = calculate_total_pages(total_records, per_page)
+    pagination = get_pagination(page, page_count)
+    return total_records, pagination
+
+
+def get_filtered_list(to_filter, filter):
+    """Filters a list based on the contents of another list"""
+    return [field for field in to_filter if field not in filter or []]
+
+
+def get_all_fields_excluding(open_search, index_name, exclude_fields=None):
+    """Retrieve all fields from the index and exclude certain fields"""
+    mappings = open_search.indices.get_mapping(index=index_name)
+    all_fields = list(mappings[index_name]["mappings"]["properties"].keys())
+    filtered_fields = get_filtered_list(all_fields, exclude_fields)
+
+    return filtered_fields
+
+
+def get_query_aggregations():
```

Comment:
nitpick but I'd suggest renaming all these query definition functions from `get_*` to `build_*`